MesaModel/model/Pupil.py

```diff
@@ -107,7 +112,10 @@ def redStateCange(self):
             self.time_in_yellow_state += 1
             self.time_in_green_state = 0
             return 1
-        # if Inattentiveness is on and quality is low
+
+        # If the pupil cohort is inattentive and the teacher cohort is of
```

Comment:
I think "pupil cohort" may be misleading here - I think `inattentiveness` and `hyper_impulsivity` are flags set by the user when running the model - so they're more like switches to say 'change state based on inattentiveness' or 'change state based on hyperimpulsivity'.

Reply:
(This applies to further similar lines below as well.)

Reply:
Hi @alisonrclarke, would you mind clarifying this for me - my reading was that these flags could be set to change the level of inattentiveness / hyperactive-impulsiveness. The above would suggest that these levels are set by the input data and the model does not change them, but rather can adjust its sensitivity to them. Is that correct?

The logic, then, would be that the significance of an inattentive class can be varied by the user not the level of inattentiveness?

Reply:
[Line 120] refers to `self.model.pupil_params.inattentiveness` which is the value passed in to the model, rather than `self.inattentiveness` which is the pupil's actual inattentiveness value. The UI only allows the pupil parameter `inattentiveness` to either 0 or 1 so it seems the intention was to use it as an on/off switch - but it's not as simple as the model being affected by a pupil's inattentiveness or not, because [line 185](https://github.com/DurhamARC/classroom-abm/blob/cd629f3445d921b45a45b3059d8c7d8446135138/MesaModel/model/Pupil.py#L185) uses the pupil's inattentiveness even when the switch is off. It may just have been a way to try out two different model specifications.